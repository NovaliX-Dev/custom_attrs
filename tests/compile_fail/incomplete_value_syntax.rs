#![allow(unused)]
use custom_attrs::CustomAttrs;

#[derive(CustomAttrs)]
#[attr(pub a: usize)]
#[attr(pub b: usize)]
enum Enum {
    #[attr(a = b =)]
    #[attr(b)]
    Variant1,

    // valid
    #[attr(a = 3)]
    #[attr(b = 3)]
    Variant2,
    
    #[attr(a = )]
    #[attr(b = 3)]
    Variant3,
}

fn main() {
    // let a = Enum::Variant1.get_a();
}
