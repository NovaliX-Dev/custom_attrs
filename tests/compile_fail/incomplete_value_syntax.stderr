error: unexpected end of input, expected expression
 --> tests/compile_fail/incomplete_value_syntax.rs:8:18
  |
8 |     #[attr(a = b =)]
  |                  ^

error: expected `=`
 --> tests/compile_fail/incomplete_value_syntax.rs:9:13
  |
9 |     #[attr(b)]
  |             ^

error: Value not set for `a`.
  --> tests/compile_fail/incomplete_value_syntax.rs:10:5
   |
10 |     Variant1,
   |     ^^^^^^^^

error: Value not set for `b`.
  --> tests/compile_fail/incomplete_value_syntax.rs:10:5
   |
10 |     Variant1,
   |     ^^^^^^^^
